src/components/side_panel/remove_duplicates/remove_duplicates.ts

```diff
@@ -0,0 +1,123 @@
+import { Component, onWillUpdateProps, useState } from "@odoo/owl";
+import { numberToLetters, zoneToDimension } from "../../../helpers";
+import { _t } from "../../../translation";
+import { HeaderIndex, SpreadsheetChildEnv } from "../../../types/index";
+import { css } from "../../helpers";
+import { RemoveDuplicateTerms } from "../../translations_terms";
+import { SidePanelErrors } from "../side_panel_errors/side_panel_errors";
+
+css/* scss */ `
+  .o-checkbox-selection {
+    height: 150px;
+  }
+`;
+
+interface Props {
+  onCloseSidePanel: () => void;
+}
+
+interface RemoveDuplicatesState {
+  hasHeader: boolean;
+  columns: { [colIndex: number]: boolean };
+}
+
+export class RemoveDuplicatesPanel extends Component<Props, SpreadsheetChildEnv> {
+  static template = "o-spreadsheet-RemoveDuplicatesPanel";
+  static components = { SidePanelErrors };
+
+  state: RemoveDuplicatesState = useState({
+    hasHeader: false,
+    columns: {},
```

Comment:
to see what others think, 

I think that using 'selected' is ambiguous because there are two notions of selection here:

- is considered selected the reference to the columns present in the selection (selection on the grid)

- and is considered selected the columns with which the user has chosen to work in the sidepanel. (columns with the checkbox selected in the side panel).

Change taken into account for the map

:)

Reply:
use a map on the UI side is annoying on the component. 
I will keep as a basic dictionary :)